src/hubAuth.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import * as os from 'os';
+import { fs } from '@salesforce/core';
+import * as shell from 'shelljs';
+import { debug } from 'debug';
+
+// this seems to be a known eslint error for enums
+// eslint-disable-next-line no-shadow
+enum AuthStrategy {
+  JWT = 'JWT',
+  AUTH_URL = 'AUTH_URL',
+  REUSE = 'REUSE',
+  NONE = 'NONE',
+}
+
+export const authFromStubbedHome = (homeDir: string): void => {
+  const logger = debug('testkit:authFromStubbedHome');
+  if (getAuthStrategy() === AuthStrategy.JWT && process.env.TESTKIT_JWT_KEY) {
+    logger('trying jwt auth');
+    const jwtKey = path.join(homeDir, 'jwtKey');
+    fs.writeFileSync(jwtKey, process.env.TESTKIT_JWT_KEY);
```

Comment:
The creation of the jwt file needs to ensure that the resulting file is in the correct format. In my case, when I stored the contents of my jwt key file in the env var, the newlines were removed from the string. Without the proper format, the jwt:grant is failing.

Reply:
I tested it by running with an existing jwt auth'd hub (so it would read the file into env and then write here), which worked.

How do I repro the error?  Just manually remove the newlines on purpose and set it?
And are you saying that we'd want to scan the env to verify the formatting?

Reply:
I took my jwt key file `pbcopy <path to file` and pasted the contents into the env var.

Reply:
I can repro that, thanks for finding.  I'm gonna leave it for now, and doc (in the quip)